src/main/kotlin/stringcalculator/NumberCharacterDistinguisher.kt

```diff
@@ -0,0 +1,37 @@
+package stringcalculator
+
+class NumberCharacterDistinguisher {
+
+    private val regexIsNumber = Regex(pattern = """^[0-9]*${'$'}""")
```

Comment:
절대로 바뀌지 않을 상수는 static으로 선언하고, Snake case 모두 대문자가 컨벤션이야. 네이밍은 나라면 아래처럼 했을거같아.

```suggestion
companion object {
    private val NUMBER_REGEX = Regex(pattern = """^[0-9]*${'$'}""")
}
```